core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -150,6 +153,11 @@ class LogManager(logDirs: Array[File],
       Exit.halt(1)
     }
 
+    if (interBrokerProtocolVersion < KAFKA_1_0_IV0 && liveLogDirs.size() < dirs.size) {
+      fatal(s"Halting broker because log dirs ${dirs.filterNot(liveLogDirs.contains(_)).mkString(",")} are offline")
```

Comment:
We should probably also mention that this is because IBP version is two old.

Reply:
Btw, we should fix the version in the upgrade notes:

```text
<li>If the <code>inter.broker.protocol.version</code> is 0.11.1 or later, a broker will now stay online to serve replicas
        on live log directories even if there are offline log directories. A log directory may become offline due to IOException
        caused by hardware failure. Users need to monitor the per-broker metric <code>offlineLogDirectoryCount</code> to check
        whether there is offline log directory. </li>
```